### PR TITLE
Bairagti code review

### PR DESCRIPTION
**Pull request by trapti1701**
Bellow is a list of 3 changes made:

There were two navigation bar, one on the index page and one separately on the home page. Moved home pages' separate navigation bar to index page to look it better. 
Logo was added to home page navigation bar, moved it to index page so that it will be visible on each page instead of only on home page. Increased consistency across pages.

Moved  the footer code from home page to index page, to make the footer visible from each page instead of only on the home page. Also changed code of the footer to make it look more good. Increased consistency across pages.

Removed CSS from each model and moved them to separate folder under CSS folder, to make other pages easy to code. 
Increased readability and structure. 

Citations:

https://www.w3schools.com/css/
https://getbootstrap.com/docs/5.2/examples/footers/

My Takeaways

While reviewing this teams code, I learned several things. Listed are the learnings :

1. Importance of Header and Footer Organisation
 
2. CSS Modularity
  
3. Importance of Documentation
 
4. Aesthetic Considerations

5. Continuous Improvement in Documentation

In summary, the code review process provided valuable insights into code organization, design principles, documentation practices, and the importance of thorough testing. Incorporating these takeaways will contribute to the team's ongoing efforts to produce high-quality, maintainable code and foster a collaborative development environment.

The teams code was present in time, however I have not made more aesthetic changes on the footer due to time constraints.

The project complies and works.

### CODE REVIEW

File: xmlproject/xmlproject/Pages/AboutYumSafe.cshtml [hunk 1]
```diff
@@ -2,40 +2,9 @@
...
-		font-family: Papyrus;
-		font-size: 150%;
-	}
 
-	p {
-		color: black;
-		font-family: Papyrus;
-		font-size: 100%;
-	}
+ <link rel="stylesheet" href="~/css/About.css" asp-append-version="true" />
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/AboutYumSafe.cshtml:6]: Good idea.

Go one step further - move this link to _Layout.cshtml

File: xmlproject/xmlproject/Pages/Index.cshtml [hunk 2]
```diff
@@ -8,186 +8,43 @@
 <head>
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Sofia&effect=fire">
     <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Sofia&effect=neon|outline|emboss|shadow-multiple">
+    <link rel="stylesheet" href="~/css/Index.css" asp-append-version="true" />
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/Index.cshtml:13]: These tags do not belong here, and should be moved to _Layout.cshtml

File: xmlproject/xmlproject/Pages/Index.cshtml [hunk 3]
```diff
@@ -8,186 +8,43 @@
...
+                       <div class="col">
+                           <input class="form-control" type="search" placeholder="Restaurants" aria-label="Search">
+                       </div>
+                       <div class="col-auto">
+                           <button class="btn btn-outline-success" type="submit">Search</button>
+                       </div>
+                   </div>
+               </form>
+           </div>
+       </nav> -->
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/Index.cshtml:47]: Avoid pushing commented code to Version Control.  Version control gives us better options:
- You can look at history to see what used to be there.
- You can use branches for experimental features.

File: xmlproject/xmlproject/Pages/Index.cshtml [hunk 4]
```diff
@@ -245,7 +102,7 @@
     </div>
     <h2></h2>
 
-    <div class="footer">
+   <!-- <div class="footer">
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/Index.cshtml:105]: Avoid pushing commented code to Version Control.  Version control gives us better options:
- You can look at history to see what used to be there.
- You can use branches for experimental features.

File: xmlproject/xmlproject/Pages/ReviewForm.cshtml [hunk 5]
```diff
@@ -48,7 +48,7 @@
 
 
             <div class="form-group">
-                <label for="directions">Tell us about your experience</label></br>
+                <label for="directions">Tell us about your experience</label>
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/ReviewForm.cshtml:51]: Yeah... this \br looks funny.  A close tag for a br, which does not have a close tag.

File: xmlproject/xmlproject/Pages/Shared/_Layout.cshtml [hunk 6]
```diff
@@ -12,7 +12,7 @@
     <header>
         <nav class="navbar navbar-expand-sm navbar-toggleable-sm navbar-light bg-white border-bottom box-shadow mb-3">
             <div class="container">
-                <a class="navbar-brand" asp-area="" asp-page="/Index">YumSafe</a>
+                 <a class="navbar-brand" asp-area="" asp-page="/Index"><img src="images/Logo.png" alt="YumSafe Logo" style="width:20%"></a>
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/Shared/_Layout.cshtml:15]: Good, with a few more adjustments:
- Describe the logo.
- Move styling to linked css.

File: xmlproject/xmlproject/Pages/Shared/_Layout.cshtml [hunk 7]
```diff
@@ -28,7 +28,19 @@
...
+                    <form class="form-inline my-2 my-lg-0">
+                                    <div class="row">
+                                        <div class="col">
+                                            <input class="form-control" type="search" placeholder="Restaurants" aria-label="Search">
+                                        </div>
+                                        <div class="col-auto">
+                                            <button class="btn btn-outline-success" type="submit">Search</button>
+                                        </div>
+                                    </div>
+                                </form>
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/Shared/_Layout.cshtml:43]: This looks like new functionality, as I don't see where this form is submitted.

Avoid adding unimplemented features, as unused code is clutter, which is technical debt.

File: xmlproject/xmlproject/Pages/Shared/_Layout.cshtml [hunk 8]
```diff
@@ -46,7 +58,20 @@
...
 
-    <script src="~/lib/jquery/dist/jquery.min.js"></script>
+<div class="container">
+   <footer class="py-3 my-4">
+     <ul class="nav justify-content-center border-bottom pb-3 mb-3">
+       <li class="nav-item"><a href="#" class="nav-link px-2 text-muted">Email: contactus@yumsafe.com</a></li>
+       <li class="nav-item"><a href="#" class="nav-link px-2 text-muted">Facebook</a></li>
+       <li class="nav-item"><a href="#" class="nav-link px-2 text-muted">Twiter</a></li>
+         <li class="nav-item"><a href="#" class="nav-link px-2 text-muted">Instagram</a></li>
+
```
Comment (discospiff) [inline xmlproject/xmlproject/Pages/Shared/_Layout.cshtml:68]: These links are incomplete.  Avoid adding unimplemented features, as unused code is clutter, which is technical debt.

File: xmlproject/xmlproject/Rating.cs [hunk 9]
```diff
@@ -14,7 +14,15 @@ namespace RatingData
     using System.Globalization;
     using Newtonsoft.Json;
     using Newtonsoft.Json.Converters;
-
+    public enum StateEnum
+    {
+        // Define the possible values for the State property
```
Comment (discospiff) [inline xmlproject/xmlproject/Rating.cs:19]: Do not put conversation in source code comments.  That's clutter.  Put the conversation here, in the Files Changed section of the pull request, where you can have a dialog with other developers.

Also, the enum is incomplete.  Avoid adding unimplemented features, as unused code is clutter, which is technical debt.

File: xmlproject/xmlproject/Rating.cs [hunk 10]
```diff
@@ -27,7 +35,7 @@ public partial class Rating
         public string City { get; set; }
 
         [JsonProperty("State")]
-        public string State { get; set; }
+        public StateEnum State { get; set; } // Used the enum type
```
Comment (discospiff) [inline xmlproject/xmlproject/Rating.cs:38]: " // Used the enum type" is a given, from the source code.  This is clutter, and should be removed.

**Comment by discospiff** [comment]
From this review, I'd harvest the linked CSS suggestion.  But, go one step further, and add them to _Layout.cshtml.

Other changes may not reduce technical debt.  See my comments in Files Changed.
